### PR TITLE
Add small fixes

### PR DESCRIPTION
**Pull request by StrawWagen**
Changed terminology from speed to accel
When testing with lfs, acceleration seemed to randomly spike in turns
Averages accel between the current and last check to fix that.
+ fixed some issues found when live testing

### CODE REVIEW

File: lua/autorun/client/cl_dfa.lua [hunk 1]
```diff
@@ -25,31 +25,30 @@ function doBlackoutOverlayEffect( alpha )
...
+    blackingOutTarget = math.Clamp( blackingOutTarget, 0, maxAlpha )
 
     if blackoutAlpha ~= blackingOutTarget then
 
         if blackoutAlpha < 1 and blackingOutTarget == 0 then -- never get stuck on near zero values
             blackoutAlpha = 0
 
         elseif blackoutAlpha < blackingOutTarget then
-            local toAdd = math.random( 18, 25 ) -- randomize this so it feel less jerky
+            local toAdd = math.random( 8, 12 ) -- randomize this so it feel less jerky
```
Comment (wrefgtzweve) [inline lua/autorun/client/cl_dfa.lua:41]: why?

File: lua/autorun/client/cl_dfa.lua [hunk 2]
```diff
@@ -25,31 +25,30 @@ function doBlackoutOverlayEffect( alpha )
 end
 
 local blackoutAlpha = 0
-local maxAlpha = 235 -- if blackoutAlpha somehow ends up above 255 then it will be stuck blacked out for like seconds, so max
+local maxAlpha = 254 -- if blackoutAlpha somehow ends up above 255 then it will be stuck blacked out for like seconds, so max
```
Comment (wrefgtzweve) [inline lua/autorun/client/cl_dfa.lua:28]: what's this for?

File: lua/autorun/client/cl_dfa.lua [hunk 3]
```diff
@@ -25,31 +25,30 @@ function doBlackoutOverlayEffect( alpha )
...
             blackoutAlpha = 0
 
         elseif blackoutAlpha < blackingOutTarget then
-            local toAdd = math.random( 18, 25 ) -- randomize this so it feel less jerky
+            local toAdd = math.random( 8, 12 ) -- randomize this so it feel less jerky
             blackoutAlpha = math.Clamp( blackoutAlpha + toAdd, 0, blackingOutTarget ) -- fast ramp up
 
         elseif blackoutAlpha > blackingOutTarget then
-            blackoutAlpha = math.Clamp( blackoutAlpha - 0.45, blackingOutTarget, maxAlpha ) -- goes down slow tho
+            blackoutAlpha = math.Clamp( blackoutAlpha - 0.2, blackingOutTarget, 255 ) -- goes down slow tho
```
Comment (wrefgtzweve) [inline lua/autorun/client/cl_dfa.lua:45]: what does this change

File: lua/autorun/server/sv_dfa.lua [hunk 4]
```diff
@@ -20,15 +20,18 @@ cvars.AddChangeCallback( "dfa_interval", function( _, _, val )
     checkInterval = tonumber( val )
 end )
 
-local function damageVehicle( veh, driver, speed )
-    local speedDiff = speed - punishSpeed
-    local damage = math.Clamp( speedDiff / 8, 0, 200 ) / 10
+local world = game.GetWorld()
```
Comment (wrefgtzweve) [inline lua/autorun/server/sv_dfa.lua]: im not sure if this will properly work before entity init

File: lua/autorun/server/sv_dfa.lua [hunk 5]
```diff
@@ -69,20 +72,27 @@ local function checkVehicle( veh, trackEnt )
...
-    local blackoutStart = warningStartOffset - punishSpeed
-    local blackoutScale = math.Clamp( speed + blackoutStart, 0, clampMagicNumber ) -- makes it get blacker faster
+    local averageAccel = ( accel + oldAccel ) / 2 -- use average so random, insane 1 tick acceleration isnt as crazy
+
+    local blackoutStart = warningStartOffset - punishAccel
+    local blackoutScale = math.Clamp( averageAccel + blackoutStart, 0, clampMagicNumber ) -- makes it get blacker faster
     local blackoutAmount = blackoutScale * ( blackoutScaleDivisor * 1.1 )
 
-    if speed > blackoutStart then
+    --driver:PrintMessage( HUD_PRINTTALK, math.Round( accel ) )
```
Comment (wrefgtzweve) [inline lua/autorun/server/sv_dfa.lua]: comment should be removed

File: lua/autorun/client/cl_dfa.lua [hunk 6]
```diff
@@ -25,31 +25,30 @@ function doBlackoutOverlayEffect( alpha )
...
     doBlackoutOverlayEffect( blackoutAlpha )
 
-    local finalAlphaClamped = math.Clamp( blackoutAlpha, 0, 250 ) -- players can always see a bit
-
-    surface_SetDrawColor( 0, 0, 0, finalAlphaClamped )
+    surface_SetDrawColor( 0, 0, 0, blackoutAlpha )
     surface_DrawRect( 0, 0, ScrW(), ScrH() )
 
-end )
+end )
```
Comment (wrefgtzweve) [inline lua/autorun/client/cl_dfa.lua]: no newline at the end of the file

File: lua/autorun/client/cl_dfa.lua (hunk 1 above)
Comment (StrawWagen) [reply lua/autorun/client/cl_dfa.lua:41]: decreased this to make it feel less jerky/flickery

File: lua/autorun/client/cl_dfa.lua (hunk 3 above)
Comment (StrawWagen) [reply lua/autorun/client/cl_dfa.lua:45]: simplifiying the clamping setup as well

File: lua/autorun/client/cl_dfa.lua (hunk 2 above)
Comment (StrawWagen) [reply lua/autorun/client/cl_dfa.lua:28]: i simplifed the clamping setup

**Comment by wrefgtzweve** [comment]
There should be an early return in the `HUDPaint` hook, no reason to run any of that code on players who aren't in vehicles / blacking out

**Comment by StrawWagen** [comment]
Untested now

**Comment by StrawWagen** [comment]
Tested.

**Review by wrefgtzweve — APPROVED** [review]
Approving on the premise that this was properly tested

**Pull request merged**